src/libcore/num/mod.rs

```diff
@@ -2536,6 +2583,19 @@ same_sign_try_from_int_impl!(i128, i128, i8, i16, i32, i64, i128, isize);
 same_sign_try_from_int_impl!(u128, usize, u8, u16, u32, u64, u128, usize);
 same_sign_try_from_int_impl!(i128, isize, i8, i16, i32, i64, i128, isize);
 
+same_sign_try_from_int_impl!(f64, usize, f32, f64);
+same_sign_try_from_int_impl!(f64, isize, f32, f64);
+same_sign_try_from_int_impl!(f64, i128, f32, f64);
+same_sign_try_from_int_impl!(f64, u128, f32, f64);
+same_sign_try_from_int_impl!(f64, i64, f32, f64);
+same_sign_try_from_int_impl!(f64, u64, f32, f64);
+same_sign_try_from_int_impl!(f64, i32, f32, f64);
+same_sign_try_from_int_impl!(f64, u32, f32, f64);
+same_sign_try_from_int_impl!(f64, i16, f32, f64);
+same_sign_try_from_int_impl!(f64, u16, f32, f64);
+same_sign_try_from_int_impl!(f64, i8, f32, f64);
+same_sign_try_from_int_impl!(f64, u8, f32, f64);
```

Comment:
For what it's worth the bounds for these conversions are wrong. NaN needs to be taken into account. Also float to int conversions round towards 0 so for example `-0.9 as u8 == 0` and `255.9 as u8 == 255` so the bounds need to allow that.